examples/list.rs

```diff
@@ -202,80 +188,159 @@ fn run_app<B: Backend>(
+    let outer_block = Block::default()
+        .borders(Borders::NONE)
+        .fg(TEXT_COLOR)
+        .bg(TODO_HEADER_BG)
+        .title("TODO List")
+        .title_alignment(Alignment::Center);
+    let inner_block = Block::default()
+        .borders(Borders::NONE)
+        .fg(TEXT_COLOR)
+        .bg(NORMAL_ROW_COLOR);
+
+    // We get the inner area from outer_block. We'll use this area later to render the table.
+    let outer_area = area[0];
+    let inner_area = outer_block.inner(outer_area);
 
-    // Iterate through all elements in the `items` app and append some debug text to it.
+    // We can render the header in outer_area.
+    f.render_widget(outer_block, outer_area);
+
+    // Iterate through all elements in the `items` and stylize them.
     let items: Vec<ListItem> = app
         .items
         .items
         .iter()
-        .map(|i| {
-            let mut lines = vec![Line::from(i.0.bold()).alignment(Alignment::Center)];
-            for _ in 0..i.1 {
-                lines.push(
-                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
-                        .italic()
-                        .into(),
-                );
-            }
-            ListItem::new(lines).style(Style::default().fg(Color::Black).bg(Color::White))
+        .enumerate()
+        .map(|(i, (todo, _info, status))| {
+            let bg_color = match i % 2 {
+                0 => NORMAL_ROW_COLOR,
+                _ => ALT_ROW_COLOR,
+            };
+            let (style, todo_str) = match status {
+                true => (
+                    Style::default().fg(COMPLETED_TEXT_COLOR).bg(bg_color),
+                    " ✓ ".to_string() + todo,
+                ),
+                false => (
+                    Style::default().fg(TEXT_COLOR).bg(bg_color),
+                    " ☐ ".to_string() + todo,
+                ),
+            };
+            ListItem::new(Line::from(todo_str)).style(style)
         })
         .collect();
 
     // Create a List from all list items and highlight the currently selected one
     let items = List::new(items)
-        .block(Block::default().borders(Borders::ALL).title("List"))
+        .block(inner_block)
         .highlight_style(
             Style::default()
-                .bg(Color::LightGreen)
-                .add_modifier(Modifier::BOLD),
+                .add_modifier(Modifier::BOLD)
+                .add_modifier(Modifier::REVERSED)
+                .fg(SELECTED_STYLE_FG),
         )
-        .highlight_symbol(">> ");
+        .highlight_symbol(">");
 
     // We can now render the item list
-    f.render_stateful_widget(items, item_list_area, &mut app.items.state);
+    f.render_stateful_widget(items, inner_area, &mut app.items.state);
+
+    let info = if let Some(i) = app.items.state.selected() {
+        match app.items.items[i].2 {
+            true => "✓ DONE: ".to_string() + app.items.items[i].1,
```

Comment:
This would work well as a enum instead of a bool. It would give you a place to hang this conversion off (impl Display perhaps, though that wouldn't give you any ability to add color or style, perhaps to_line()?)

Reply:
We need two different strings for todo list and todo info so impl Display won't work imo. But I agree with the opinion of using an enum.